### PR TITLE
[4.4] Language Filter strings

### PR DESCRIPTION
**Pull request by brianteeman**
### Summary of Changes

With #36748 

> "This PR checks if a translation for the language key exists and if there is none (and debug is not enabled) it will instead display the taxonomy title instead."

It is confusing that this doesnt work if debug is enabled. There is nothing else in Joomla where this is the case and I just wasted a lot of time trying to see why the string wasnt displayed.

I believe that the **_intention_** of the code was to display the language key when "debug language" was enabled not when "debug" was enabled

If you want to create your own value for the key as an override you would expect the key to be displayed in debug language mode (which it is not) and you would not expect the key to be displayed in debug mode 
### Testing Instructions
Perform a search on the front with Debug enabled and a new search with debug language enabled

#### Debug
image

#### Debug Language
image

#### Debug
image

#### Debug Language
image

### Link to documentations
Please select:
- [ ] Documentation link for docs.joomla.org: <link>
- [x] No documentation changes for docs.joomla.org needed

- [ ] Pull Request link for manual.joomla.org: <link>
- [x] No documentation changes for manual.joomla.org needed

### CODE REVIEW

File: administrator/components/com_finder/src/Helper/LanguageHelper.php [hunk 1]
```diff
@@ -58,8 +58,9 @@ public static function branchSingular($branchName)
     {
         $return   = preg_replace('/[^a-zA-Z0-9]+/', '_', strtoupper($branchName));
         $language = Factory::getApplication()->getLanguage();
+        $debug = Factory::getApplication()->get('debug_lang');
```
Comment (richard67) [inline administrator/components/com_finder/src/Helper/LanguageHelper.php]:
```suggestion
        $debug    = Factory::getApplication()->get('debug_lang');
```
Fix php-cs-fixer failing in Drone.

**Comment by Quy** [comment]
I have tested this item :white_check_mark: successfully on f0ee89d578f3aba098c1ed7230cd0d75a8359caf<hr /><sub>This comment was created with the <a href="https://github.com/joomla/jissues">J!Tracker Application</a> at <a href="https://issues.joomla.org/tracker/joomla-cms/42687">issues.joomla.org/tracker/joomla-cms/42687</a>.</sub>

**Comment by fgsw** [comment]
I have tested this item :white_check_mark: successfully on f0ee89d578f3aba098c1ed7230cd0d75a8359caf<hr /><sub>This comment was created with the <a href="https://github.com/joomla/jissues">J!Tracker Application</a> at <a href="https://issues.joomla.org/tracker/joomla-cms/42687">issues.joomla.org/tracker/joomla-cms/42687</a>.</sub>

**Comment by Quy** [comment]
RTC<hr /><sub>This comment was created with the <a href="https://github.com/joomla/jissues">J!Tracker Application</a> at <a href="https://issues.joomla.org/tracker/joomla-cms/42687">issues.joomla.org/tracker/joomla-cms/42687</a>.</sub>

**Comment by MacJoom** [comment]
Thank you!

**Comment by brianteeman** [comment]
thanks

**Pull request merged**